### PR TITLE
Add weather heatmap example.

### PR DESCRIPTION
**Pull request by Saba9**
Part of issue #1486. Heatmap is almost identical to the [one found in the Altair user guide](https://altair-viz.github.io/user_guide/transform.html#timeunit-transform) except it uses the San Francisco data rather than the Seattle data.

image

### CODE REVIEW

File: _data/examples.json [hunk 1]
```diff
@@ -196,6 +196,10 @@
         "title": "Table Heatmap",
         "style": "background-size: auto"
       },
+      {
+        "name": "rect_heatmap_weather",
+        "title": "Table Heatmap of Weather"
```
Comment (kanitw) [inline _data/examples.json]: I wonder if we should call this "Annual Weather Heatmap"?

(We can keep file name the same.)

Comment (Saba9) [reply _data/examples.json]: That sounds good to me. I can go ahead and update the title.

**Comment by g3o2** [comment]
maybe the gridlines could be made slightly thinner?

**Comment by kanitw** [comment]
> maybe the gridlines could be made slightly thinner?

These are actuall gap between rect bands.

@saba9 Perhaps remove the top level width and height and set rangeStep of x and y scales to the same values so we get square rects with equal gap (more beautiful)

**Pull request merged**